src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,117 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props>= ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCounter] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const requiredField = {
+    title,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  }
+
+  const isRequiered = (field: string) => {
+    return Object.keys(requiredField).includes(field)
+  }
+
+  const isRequiredSetFilled = () => {
+    return !Object.values(requiredField).some(str => {
+      return str.trim() === ''
+    })
+  }
+
+  const reset = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  }
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!isRequiredSetFilled()) {
+      return
+    }
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    })
+
+    setCounter((c) => c + 1);
+
+    reset()
+  }
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
-        required
+        value={title}
+        onChange={(e) => setTitle(e)}
```

Comment:
```suggestion
        onChange={setTitle}
```

The same in all places